packages/flutter_test/test/finders_test.dart

```diff
@@ -390,6 +395,391 @@ void main() {
+      ), findsOneWidget);
+    });
+
+    testWidgets('finds two matching ancestors, one descendant', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const Directionality(
+          textDirection: TextDirection.ltr,
+          child: Row(
+            children: <Widget>[
+              Row(children: fooBarTexts),
+            ],
+          ),
+        ),
+      );
+
+      expect(find.ancestor(
+        of: find.text('bar'),
+        matching: find.byType(Row),
+      ), findsNWidgets(2));
+    });
+
+    testWidgets('fails with a descriptive message', (WidgetTester tester) async {
+      await tester.pumpWidget(const Row(
+        textDirection: TextDirection.ltr,
+        children: <Widget>[
+          Column(children: <Text>[Text('foo', textDirection: TextDirection.ltr)]),
+          Text('bar', textDirection: TextDirection.ltr),
+        ],
+      ));
+
+      late TestFailure failure;
+      try {
+        expect(find.ancestor(
+          of: find.text('bar'),
+          matching: find.widgetWithText(Column, 'foo'),
+        ), findsOneWidget);
+      } on TestFailure catch (e) {
+        failure = e;
+      }
+
+      expect(failure, isNotNull);
+      expect(
+        failure.message,
+        contains(
+          'Actual: _AncestorFinder:<Found 0 Elements with that is exactly typed as "Column" that are ancestors of root Element(s) with text "foo" that are ancestors of root Element(s) with text "bar"',
+        ),
+      );
+    });
+
+    testWidgets('Root not matched by default', (WidgetTester tester) async {
+      await tester.pumpWidget(const Row(
+        textDirection: TextDirection.ltr,
+        children: <Widget>[
+          Column(children: fooBarTexts),
+        ],
+      ));
+
+      expect(find.ancestor(
+        of: find.byType(Column),
+        matching: find.widgetWithText(Column, 'foo'),
+      ), findsNothing);
+    });
+
+    testWidgets('Match the root', (WidgetTester tester) async {
+      await tester.pumpWidget(const Row(
+        textDirection: TextDirection.ltr,
+        children: <Widget>[
+          Column(children: fooBarTexts),
+        ],
+      ));
+
+      expect(find.descendant(
+        of: find.byType(Column),
+        matching: find.widgetWithText(Column, 'foo'),
+        matchRoot: true,
+      ), findsOneWidget);
+    });
+
+    testWidgets('is fast in deep tree', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: _deepWidgetTree(
+            depth: 1000,
+            child: Row(
+              children: <Widget>[
+                _deepWidgetTree(
+                  depth: 1000,
+                  child: const Column(children: fooBarTexts),
+                ),
+              ],
+            ),
+          ),
+        ),
+      );
+
+      expect(find.ancestor(
+        of: find.text('bar'),
+        matching: find.byType(Row),
+      ), findsOneWidget);
+    });
+  });
+
```

Comment:
To review the changes to the tests above the new `CommonSemanticsFinders` tests, [this commit](https://github.com/flutter/flutter/pull/127137/commits/7b63da59cb33741404de4efe601bcd630c2b0a0e) has _just_ those tests' updates in it.